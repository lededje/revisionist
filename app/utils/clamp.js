import moment from 'moment';

export const clampTime = (time, past, future) => moment.max(moment.min(time, future), past);
export const clampPeriod = (time, period) => clampTime(time, moment(time).startOf(period), moment(time).endOf(period));
