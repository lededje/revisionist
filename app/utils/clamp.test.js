import { clampPeriod } from './clamp';

describe('clampPeriod', () => {
  it.each([
    ['2019-05-27T15:21:59+01:00', 'day'],
    ['2019-05-27T15:21:59+01:00', 'year'],
    ['2019-05-27T15:21:59+01:00', 'hour'],
    ['2019-05-27T15:21:59+01:00', 'minute'],
  ])('clamps date %s to increment %s', (date, period) => {
    expect(clampPeriod(date, period)).toMatchSnapshot();
  });
});
