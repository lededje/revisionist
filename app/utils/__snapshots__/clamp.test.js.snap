// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clampPeriod clamps date 2019-05-27T15:21:59+01:00 to increment day 1`] = `"2019-05-26T23:00:00.000Z"`;

exports[`clampPeriod clamps date 2019-05-27T15:21:59+01:00 to increment hour 1`] = `"2019-05-27T14:00:00.000Z"`;

exports[`clampPeriod clamps date 2019-05-27T15:21:59+01:00 to increment minute 1`] = `"2019-05-27T14:21:00.000Z"`;

exports[`clampPeriod clamps date 2019-05-27T15:21:59+01:00 to increment year 1`] = `"2019-01-01T00:00:00.000Z"`;
